freeswitch_click2dial/freeswitch_click2dial.py

```diff
@@ -0,0 +1,420 @@
+    '''FreeSWITCH server object, stores the parameters of the FreeSWITCH Servers'''
+    _name = "freeswitch.server"
+    _description = "FreeSWITCH Servers"
+    name = fields.Char(string='FreeSWITCH Server Name', required=True)
+    active = fields.Boolean(
+        string='Active', help="The active field allows you to hide the "
+        "FreeSWITCH server without deleting it.", default=True)
+    ip_address = fields.Char(
+        string='FreeSWITCH IP address or DNS', required=True,
+        help="IP address or DNS name of the FreeSWITCH server.")
+    port = fields.Integer(
+        string='Port', required=True, default=8021,
+        help="TCP port on which the FreeSWITCH Event Socket listens. "
+        "Defined in "
+        "/etc/freeswitch/autoload_configs/event_socket.conf.xml on "
+        "FreeSWITCH.")
+    out_prefix = fields.Char(
+        string='Out Prefix', size=4, help="Prefix to dial to make outgoing "
+        "calls. If you don't use a prefix to make outgoing calls, "
+        "leave empty.")
+    password = fields.Char(
+        string='Event Socket Password', required=True,
+        help="Password that OpenERP will use to communicate with the "
+        "FreeSWITCH Event Socket. Refer to "
+        "/etc/freeswitch/autoload_configs/event_socket.conf.xml "
+        "on your FreeSWITCH server.")
+    context = fields.Char(
+        string='Dialplan Context', required=True, default="XML default",
+        help="FreeSWITCH dialplan context from which the calls will be "
+        "made; e.g. 'XML default'. Refer to /etc/freeswitch/dialplan/* "
+        "on your FreeSWITCH server.")
+    wait_time = fields.Integer(
+        string='Wait Time (sec)', required=True,
+        help="Amount of time (in seconds) FreeSWITCH will try to reach "
+        "the user's phone before hanging up.", default=60)
+    alert_info = fields.Char(
+        string='Alert-Info SIP Header',
+        help="Set Alert-Info header in SIP request to user's IP Phone "
+        "for the click2dial feature. If empty, the Alert-Info header "
+        "will not be added. You can use it to have a special ring tone "
+        "for click2dial (a silent one !) or to activate auto-answer "
+        "for example.")
+    company_id = fields.Many2one(
+        'res.company', string='Company',
+        default=lambda self: self.env['res.company']._company_default_get(
+            'freeswitch.server'),
+        help="Company who uses the FreeSWITCH server.")
+
+    @api.multi
+    @api.constrains(
+        'out_prefix', 'wait_time', 'port', 'context', 'alert_info', 'password')
+    def _check_validity(self):
+        for server in self:
+            out_prefix = ('Out prefix', server.out_prefix)
+            dialplan_context = ('Dialplan context', server.context)
+            alert_info = ('Alert-Info SIP header', server.alert_info)
+            password = ('Event Socket password', server.password)
+
+            if out_prefix[1] and not out_prefix[1].isdigit():
```

Comment:
On 08/26/2016 01:20 AM, Eric Caudal wrote:
>
> In freeswitch_click2dial/freeswitch_click2dial.py
> <https://github.com/OCA/connector-telephony/pull/101#discussion_r76374295>:
>
> > +        'res.company', string='Company',
> > +        default=lambda self: self.env['res.company']._company_default_get(
> > +            'freeswitch.server'),
> > +        help="Company who uses the FreeSWITCH server.")
> > +
> > +    @api.multi
> > +    @api.constrains(
> > +        'out_prefix', 'wait_time', 'port', 'context', 'alert_info', 'password')
> > +    def _check_validity(self):
> > +        for server in self:
> > +            out_prefix = ('Out prefix', server.out_prefix)
> > +            dialplan_context = ('Dialplan context', server.context)
> > +            alert_info = ('Alert-Info SIP header', server.alert_info)
> > +            password = ('Event Socket password', server.password)
> > +
> > +            if out_prefix[1] and not out_prefix[1].isdigit():
> some pointers here:
> http://www.odoo.com/documentation/9.0/howtos/backend.html#model-constraints
>
Alright. Thank you for all of your help. This is the only suggested
change that I have not yet made. (Other than a not working -> TODO
change, which I explained).

I will look at this one later today.

Thank you.
Trever